### DISCUSSION TITLE
Welcome to obsidian-lab Discussions!

### DISCUSSION THREAD

**Discussion by cristianvasquez** [cristianvasquez/obsidian-lab#1] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

**Comment by zacKolton** [comment]
Hi, love this btw, have you seen [Dual](https://psionica.org/docs/workshop/dual/)? Im wondering what the difference is between your plugin and Dual. Obviously not looking for specifics, but it seems Dual tries to accomplish the same things as yours (at least with the chat function). 

### Questions about the chat feature 
- How extensive is the chat, like for instance, is it "trainable"? 
- What does it _do_ exactly?
- Is it able to handle open dialogue like, "What do you know about me?" or just specific questions about your obsidian vault like, for instance, "where are my notes on [xyz]"?

### Other Questions
- What are your future plans/roadmap for this plugin (sorry if I missed it somewhere)?
- Is there anyway you could make a more comprehensive installation guide? Im confused on the difference between this repo and the obsidian-lab-py repo

Thanks!

**Reply by zacKolton** [reply]
After going through the README again, I suppose im wondering...

Is this plugin only useful if you have your _own_ scripts / downloaded ones? Its seems like thats the purpose, but the chat feature threw me off as it seems like its built in.

I guess what im wondering is: What is built in and what is not?

Thanks!

**Reply by cristianvasquez** [reply]
Hello Zac!. 

The Obsidian-lab is just a collection of dumb interfaces to enable python developers to use/test functionality within obsidian. I do not provide any implementation for the chat or anything else. I wrote it to test my scripts, but I would love to share scripts with others. 

In summary, what's built-in: The interfaces. what is not: The implementations.

I gave a quick look to dual when it was announced; in fact, I borrowed the 'chat' looks from there. I wanted to program my little personal bot to search things in my vault but never had the time.

**Reply by cristianvasquez** [reply]
If you are interested in doing scripts with this, just ask whatever you want
